scripts/zones/Batallia_Downs/Zone.lua

```diff
@@ -4,11 +4,76 @@
 --
 -----------------------------------
 package.loaded[ "scripts/zones/Batallia_Downs/TextIDs"] = nil;
+package.loaded["scripts/globals/chocobo_digging"] = nil;
 -----------------------------------
 
 require( "scripts/zones/Batallia_Downs/TextIDs");
 require("scripts/globals/zone");
 require( "scripts/globals/icanheararainbow");
+require("scripts/globals/chocobo_digging");
+
+-----------------------------------
+-- Chocobo Digging vars
+-----------------------------------
+local itemMap = {
+                    -- itemid, abundance, requirement
+                    { 847, 69, DIGREQ_NONE },
+                    { 880, 137, DIGREQ_NONE },
+                    { 845, 4, DIGREQ_NONE },
+                    { 640, 82, DIGREQ_NONE },
+                    { 768, 133, DIGREQ_NONE },
+                    { 643, 82, DIGREQ_NONE },
+                    { 17296, 137, DIGREQ_NONE },
+                    { 774, 26, DIGREQ_NONE },
+                    { 106, 69, DIGREQ_NONE },
+                    { 4449, 3, DIGREQ_NONE },
+                    { 4096, 100, DIGREQ_NONE },
+                    { 4097, 100, DIGREQ_NONE },
+                    { 4098, 100, DIGREQ_NONE },
+                    { 4099, 100, DIGREQ_NONE },
+                    { 4100, 100, DIGREQ_NONE },
+                    { 4101, 100, DIGREQ_NONE },
+                    { 4102, 100, DIGREQ_NONE },
+                    { 4103, 100, DIGREQ_NONE },
+                    { 656, 106, DIGREQ_BURROW },
+                    { 748, 8, DIGREQ_BURROW },
+                    { 749, 30, DIGREQ_BURROW },
+                    { 750, 136, DIGREQ_BURROW },
+                    { 1237, 30, DIGREQ_BORE },
+                    { 2235, 60, DIGREQ_BORE },
+                    { 2364, 150, DIGREQ_BORE },
+                    { 4570, 10, DIGREQ_MODIFIER },
+                    { 4487, 11, DIGREQ_MODIFIER },
+                    { 4409, 12, DIGREQ_MODIFIER },
+                    { 1188, 10, DIGREQ_MODIFIER },
+                    { 4532, 12, DIGREQ_MODIFIER },
+                };
+
+local messageArray = { DIG_THROW_AWAY, FIND_NOTHING, ITEM_OBTAINED };
+
+-----------------------------------
+-- onChocoboDig
+-----------------------------------
+function onChocoboDig(player, precheck)
+
+    -- Let's get the weather of the zone
+
+    local weather = player:getWeather();
+
+    if (weather ~= nil) then

```

Comment:
why did you do this check in every single zone instead of just getting the weather inside chocoboDig?